src/inmanta/env.py

```diff
@@ -182,26 +237,207 @@ def _run_command_and_log_output(
             LOGGER.debug("%s: %s", cmd, output.decode())
             return output.decode()
 
+    def _parse_line(self, req_line: str) -> Tuple[Optional[str], str]:
+        """
+        Parse the requirement line
+        """
+        at = VirtualEnv._at_fragment_re.search(req_line)
+        if at is not None:
+            d = at.groupdict()
+            return d["name"], d["req"] + "#egg=" + d["name"]
+
+        egg = VirtualEnv._egg_fragment_re.search(req_line)
+        if egg is not None:
+            d = egg.groupdict()
+            return d["name"], req_line
+
+        return None, req_line
+
+    def _gen_content_requirements_file(self, requirements_list: List[str]) -> str:
```

Comment:
I assume this complex logic is moved from somewhere else?

Reply:
Yes. It was moved higher up in the inheritance hierarchy (From `VirtualEnv` to `PythonEnvironment`).